prov/shm/src/smr_ep.c

```diff
@@ -1800,7 +1801,14 @@ int smr_endpoint(struct fid_domain *domain, struct fi_info *info,
 	ep->util_ep.ep_fid.msg = &smr_msg_ops;
 	ep->util_ep.ep_fid.tagged = &smr_tag_ops;
 
-	ep->cmd_ctx_fs = smr_cmd_ctx_fs_create(info->rx_attr->size, NULL, NULL);
+	ret = ofi_bufpool_create(&ep->cmd_ctx_pool, sizeof(struct smr_cmd_ctx),
+				 16, 0, info->rx_attr->size,
+				 OFI_BUFPOOL_NO_TRACK);
+	if (ret) {
+		FI_WARN(&smr_prov, FI_LOG_EP_CTRL,
+			"Unable to create cmd ctx pool\n");
+		return ret;
+	}
```

Comment:
I suggest call `ofi_bufpool_grow` here, so every endpoint start with 1 chunk of `cmd_ctx`.

Otherwise, different endpoint allocate memory at different time, which can cause overall delay.

Reply:
Hmm, I thought alloc called grow on initialization but I guess I'm wrong. I will add. Thanks!

Reply:
It is actually lazy initialization.